src/Workspaces/Core/Portable/NamingStyles/Serialization/SerializableNamingStylePreferencesInfo.cs

```diff
@@ -0,0 +1,434 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Simplification;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
+{
+    /// <summary>
+    /// Contains all information related to Naming Style Preferences.
+    /// 1. Symbol Specifications
+    /// 2. Name Style
+    /// 3. Naming Rule (points to Symbol Specification IDs)
+    /// </summary>
+    internal class SerializableNamingStylePreferencesInfo : IEquatable<SerializableNamingStylePreferencesInfo>
+    {
+        public List<SymbolSpecification> SymbolSpecifications;
+        public List<NamingStyle> NamingStyles;
+        public List<SerializableNamingRule> NamingRules;
+        private readonly static int s_serializationVersion = 3;
```

Comment:
@dpoeschl should I rev the version here?  what are the conditions under which we change this number?